### DISCUSSION TITLE
workspace symbol picker does not work with golang?

### DISCUSSION THREAD

**Discussion by wgqimut** [helix-editor/helix#5056] [Q&A] [answered]
I just download gopls, and everything is ok when I execute `hx --health go`, but the workspace symbol picker window don't show up when I open a golang project's go file and `Space+S`

I open hx with `-vvvv`, the log show the workspace/symbol result is `null`, I don't have any configure in `language.toml`,
```
2022-12-07T17:34:35.492 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2022-12-07T17:34:35.751 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2022-12-07T17:34:35.751 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"workspace/symbol","params":{"query":""},"id":2}
2022-12-07T17:34:35.752 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","result":null,"id":2}
```

the gopls version is the latest(0.10.1), helix is the latest(22.12), my OS is macOS 13.0.1, should I do anything about the golang config?

**Comment by archseer** [comment, marked as answer]
Probably fixed by https://github.com/helix-editor/helix/pull/5055

**Reply by wgqimut** [reply]
I tried this PR, it still does not work. Actually, this PR try to solve the issue which the result is `[]`, not `null`

**Reply by the-mikedavis** [reply]
Does it work if you type in a query? I think gopls sends null if the query is empty (and there isn't really anything we can do about that)

**Reply by the-mikedavis** [reply]
See here where gopls returns nil for an empty query: https://github.com/golang/tools/blob/f3fb218e75320162ef00d0678e4de7bbe3e1a1d4/gopls/internal/lsp/source/workspace_symbol.go#L58-L60

**Reply by wgqimut** [reply]
I can't input anything to query because the picker window didn't pop up, nothing happend when I press `Space+S`, that's why I check the log

**Reply by the-mikedavis** [reply]
Oh I see, we aren't opening the picker if we get None from the server. In these cases I think we should default to the empty Vec on None. I'll push a commit to #5055.

**Comment by lanlanlan3** [comment]
same to python: language server does not support workspace symbols

**Reply by the-mikedavis** [reply]
`pylsp` does not support workspace symbols at all (see https://github.com/python-lsp/python-lsp-server#lsp-server-features), so if you are running 22.12 you should see an error in the statusline saying "Language server does not support workspace symbols". This is a different issue: `gopls` does support workspace symbols but doesn't support the empty query
